endpoints/setuid.go

```diff
@@ -101,6 +103,21 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 			return
 		}
 
+		activities, activitiesErr := privacy.NewActivityControl(account.Privacy)
+		if activitiesErr != nil {
+			if errortypes.ContainsFatalError([]error{activitiesErr}) {
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
+		}
+
+		userSyncActivityAllowed := activities.Allow(privacy.ActivitySyncUser,
+			privacy.ScopedName{Scope: privacy.ScopeTypeBidder, Name: bidderName})
+		if userSyncActivityAllowed == privacy.ActivityDeny {
+			w.WriteHeader(http.StatusUnavailableForLegalReasons)
+			return
```

Comment:
We should probably record a metric and update the analytics object `so` here as well.
For GDPR, the metric we use when the error is `http.StatusUnavailableForLegalReasons` is `metrics.SetUidGDPRHostCookieBlocked`. Maybe we need to add a more generic privacy related blocking metric like `metrics.SetUidPrivacyHostCookieBlocked`?

Reply:
We discussed this offline and decided to add metrics in a separate PR.